src/components/multi-select/multi-select.jsx

```diff
@@ -0,0 +1,316 @@
+        labelId={ids.label}
+        id={ids.textbox}
+        error={validationMessage}
+        onKeyDown={onKeyDown}
+      >
+        <div role="presentation" className={classNames.formControlChildrenContainer} onClick={onClick}>
+          <TagList
+            className={classNames.tagList}
+            labelledBy={ids.label}
+            refs={valueRefs}
+          >
+            {value.length !== 0 &&
+              props.tagChildren ?
+              props.tagChildren(value, valueRefs, onOptionRemove) :
+              value.map((val, index) => (
+                <Tag
+                  defaultActive={index === 0}
+                  id={`${props.id}-option-${props.optionIDGetter(getOption(val))}`}
+                  key={props.optionIDGetter(getOption(val))}
+                  onRemove={onOptionRemove}
+                  readOnly={props.readOnly}
+                  ref={valueRefs[index]}
+                >
+                  {props.optionLabelGetter(
+                    props.options.find(option => props.optionIDGetter(option) === props.optionIDGetter(getOption(val))),
+                  )}
+                </Tag>
+              ))
+            }
+            <input
+              aria-autocomplete="list"
+              aria-controls={ids.listbox}
+              aria-describedby={`${ids.emptyMessage}`}
+              aria-expanded={expanded}
+              aria-haspopup="listbox"
+              aria-labelledby={ids.label}
+              autoComplete="off"
+              role="combobox"
+              className={classNames.input}
+              id={ids.textbox}
+              onBlur={onAutocompleteBlur}
+              onChange={onAutocompleteChange}
+              placeholder={value.length === 0 ? props.placeholder : undefined}
+              readOnly={props.readOnly}
+              required={props.required ? value.length === 0 : false}
+              ref={autocompleteRef}
+              value={autocompleteValue}
+            />
+          </TagList>
+          <FormControlIcons validationMessage={props.validationMessage} className={classNames.iconsContainer}>
+            {value.length > 0 && (
+              <IconButton
+                icon={iconClear}
+                label={`Remove all selected options on ${props.label}`}
+                onPress={onOptionsClear}
+                className={classNames.iconClear}
+              />
+            )}
+            <Icon icon={iconCaretDown} label="Open multi select options" />
+          </FormControlIcons>
+        </div>
+        {dropdownContents()}
+      </FormControl>
+    </div>
+  );
+});
+
+MultiSelect.propTypes = {
+  classNames: PropTypes.shape({
+    container: PropTypes.string,
+    formControlChildrenContainer: PropTypes.string,
+    input: PropTypes.string,
+    tagList: PropTypes.string,
+  }),
+  id: PropTypes.string.isRequired,
+  label: PropTypes.string.isRequired,
+  listboxChildren: PropTypes.func, // eslint-disable-line react/no-unused-prop-types
```

Comment:
Note: These are ignored because the linter doesn't see them being used. Check out `dropdownContents` on line 57.

Reply:
@cbacon91 I recall you fixed something like recently. Do you know how to fix these? I couldn't figure it out -- is our eslint just outta date or something? 🤔

Reply:
It might have to do with the fact that the prop is used in a "internal function component" so maybe it can't correlate the usage of props in there with the parent's closure. smh linters.

Reply:
I have the same feeling. Can't differentiate separate function scopes sometimes, for some reason. 😬

Reply:
if you want, you can extract the "internal component" into a private component in this file. I usually wrap it in `/* eslint-disable prop-types-stuff */` and do not write tests (since the tests are on the actual exported component). It might "clean up" the component too?

Reply:
Not sure I love this option. I think it unnecessarily complicates the flow by introducing more props passing, creates additional points of failure, and complicates the file by putting two component declarations in the same place. We still also need to use `eslint-disable`, so it also doesn't dig us out of that hole. 😬

Reply:
> @cbacon91 I recall you fixed something like recently.

I didn't fix anything, I just removed some rules that weren't needed.